zipline/lib/_windowtemplate.pxi

```diff
@@ -84,7 +86,9 @@ cdef class AdjustedArrayWindow:
         # Apply any adjustments that occured before our current anchor.
         # Equivalently, apply any adjustments known **on or before** the date
         # for which we're calculating a window.
-        while self.next_adj < anchor:
+        while (self.next_adj != -1
```

Comment:
I think this might be more clearly expressed as `while self.next_adj < anchor + self.perspective_offset`.